integration-tests/integration_test.go

```diff
@@ -828,6 +834,110 @@ func (s *ProcessListeningOnPortTestSuite) TestProcessListeningOnPort() {
 	assert.True(s.T(), hasOpenAndClose(endpoints9091), "Did not capture open and close events for port 9091")
 }
 
+func (s *SocatTestSuite) SetupSuite() {
+
+	s.metrics = map[string]float64{}
+	s.executor = NewExecutor()
+	s.StartContainerStats()
+	s.collector = NewCollectorManager(s.executor, s.T().Name())
+
+	s.collector.Env["COLLECTOR_CONFIG"] = `{"logLevel":"debug","turnOffScrape":false,"scrapeInterval":2}`
+	s.collector.Env["ROX_PROCESSES_LISTENING_ON_PORT"] = "true"
+
+	err := s.collector.Setup()
+	s.Require().NoError(err)
+
+	err = s.collector.Launch()
+	s.Require().NoError(err)
+	time.Sleep(30 * time.Second)
+
+	processImage := qaImage("quay.io/rhacs-eng/qa", "socat")
+
+	containerID, err := s.launchContainer("socat", processImage, "/bin/sh", "-c", "socat TCP-LISTEN:80,fork STDOUT & socat TCP-LISTEN:8080,fork STDOUT")
+
+	s.Require().NoError(err)
+	s.serverContainer = containerShortID(containerID)
+
+	time.Sleep(6 * time.Second)
+
+	err = s.collector.TearDown()
+	s.Require().NoError(err)
+
+	s.db, err = s.collector.BoltDB()
+	s.Require().NoError(err)
+}
+
+func (s *SocatTestSuite) TearDownSuite() {
+	s.cleanupContainer([]string{"socat", "collector"})
+	stats := s.GetContainerStats()
+	s.PrintContainerStats(stats)
+	s.WritePerfResults("Socat", stats, s.metrics)
+}
+
+func getEndpointByPort(endpoints []EndpointInfo, port int) (*EndpointInfo, error) {
+	for _, endpoint := range endpoints {
+		if endpoint.Address.Port == port {
+			return &endpoint, nil
+		}
+	}
+
+	err := fmt.Errorf("Could not find endpoint with port %d", port)
+
+	return nil, err
+}
+
+func getProcessByPort(processes []ProcessInfo, port int) (*ProcessInfo, error) {
+	re := regexp.MustCompile(`:(` + strconv.Itoa(port) + `),`)
+	for _, process := range processes {
+		portArr := re.FindStringSubmatch(process.Args)
+		if len(portArr) == 1 {
```

Comment:
```suggestion
		if len(portArr) == 2 {
```
`FindStringSubmatch` returns the whole match, followed by submatches.